src/java/org/apache/cassandra/hints/HintsServiceMBean.java

```diff
@@ -50,4 +50,17 @@ public interface HintsServiceMBean
      * @return a list of endpoints with relevant hint information - total number of files, newest and oldest timestamps.
      */
     List<Map<String, String>> getPendingHints();
+
+    /**g
+     * Gets total size of all hints for a respective node
+     * @param address address of a node to get size of hints for
+     * @return size of hints for a given address
+     */
+    long getHintsSizeForEndpoint(String address);
```

Comment:
When one checks this more closely, there is only resolution of nodes on address level, if there are two nodes behind same address (but on different ports), then this does not work. HintsCatalog and HintsStore are (host)-UUID based but these methods are not.

This does not work already for `deleteAllHintsForEndpoint`. 

Ideally, there should be these methods added too:

`deleteAllHintsForNodeId`
`getHintsSizeForNodeId`

We might leave getHintsSize(String address)